### PR TITLE
[BUGFIX] Respect categories from demand

### PR DESCRIPTION
**Pull request by develth**
Fixes issue that categories set by overwriteDemand get lost

### CODE REVIEW

File: Classes/Controller/CategoryController.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ public function listAction(array $overwriteDemand = null)
             $demand = $this->overwriteDemandObject($demand, $overwriteDemand);
         }
 
-        $idList = explode(',', $this->settings['categories']);
+        $idList = is_array($demand->getCategories())?$demand->getCategories():explode(',', $demand->getCategories());
```
Comment (DavidBruchmann) [inline Classes/Controller/CategoryController.php:33]:
```suggestion
        $idList = is_array($demand->getCategories()) ? $demand->getCategories() : explode(',', $demand->getCategories());
```

Comment (DavidBruchmann) [reply Classes/Controller/CategoryController.php:33]: I never checked functionality, it's just about the syntax.

**Review by georgringer — APPROVED** [review]

**Comment by Megafry** [comment]
Hi
I have a list with several categories, ex:
_Categories 1, Categories 2, Categories 3_

With the old code I could set on as active, ex: 
_Categories 1, **Categories 2**, Categories 3_

Unfortunately this is no longer possible because I only get one category, ex:
**Categories 2**

Could this option be enabled / disabled via TypoScript ?

**Comment by georgringer** [comment]
@Megafry done with 7ae3e1c0

**Comment by develth** [comment]
Hi,
so currently there is no way to overwrite categories within filter in e.g. an ajax request?

**Pull request merged**